cli/tests/test_mcp_dns_pinning.py

```
@@ -0,0 +1,279 @@
+    [
+        pytest.param("ollama", True, id="local-provider"),
+        pytest.param("anthropic", False, id="cloud-provider"),
+    ],
+)
+def test_only_local_provider_default_can_resolve_loopback(
+    provider,
+    allows_loopback_default,
+):
+    calls: list[str] = []
+
+    def fake_getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `fake_getaddrinfo`

(ANN202)